src/meta/src/rpc/service/ddl_service.rs

```diff
@@ -241,11 +247,19 @@ where
         &self,
         request: Request<DropSinkRequest>,
     ) -> Result<Response<DropSinkResponse>, Status> {
-        let sink_id = request.into_inner().sink_id;
-
+        let request = request.into_inner();
+        let sink_id = request.sink_id;
+        let drop_mode = if request.cascade {
+            DropMode::Cascade
+        } else {
+            DropMode::Restrict
+        };
```

Comment:
Saw it being used in a few areas, can it be abstracted to:
`let drop_mode = DropMode::from_request_setting(request.cascade)`